### PR TITLE
NEWRELIC-3911 make the e2e run locally

### PR DESCRIPTION
**Pull request by marcsanmi**
Fix e2e to make them run locally.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 NRI_GENERATOR_PATH="$(PWD)/nri-config-generator"
 
+GOOS ?=
+GOARCH ?=
+
 ifeq (, $(shell which newrelic-integration-e2e))
-NEWRELIC_E2E ?= go run github.com/newrelic/newrelic-integration-e2e-action/newrelic-integration-e2e/cmd@v1
+NEWRELIC_E2E ?= go run github.com/newrelic/newrelic-integration-e2e-action/newrelic-integration-e2e/cmd@latest
 else
 NEWRELIC_E2E ?= newrelic-integration-e2e
```
Comment (paologallinaharbur) [inline Makefile:10]: does it make sense to keep the installed command? I fear this can create issues

File: Makefile [hunk 2]
```diff
@@ -42,11 +47,11 @@ package-%:
 	bash scripts/copy_resources.sh $(PWD) $* && \
 	bash scripts/package.sh $(PWD) $*
 
-test-e2e-%: 
+test-e2e-%: build-%
 	@echo "[ test-e2e-%$* ]: Running e2e test..."
 	$(NEWRELIC_E2E) --commit_sha=test-string --retry_attempts=5 --retry_seconds=60 \
          --account_id=$(ACCOUNT_ID) --api_key=$(API_KEY) --license_key=$(LICENSE_KEY) \
-         --spec_path=$(PWD)/exporters/$*/e2e/e2e_spec.yml --verbose_mode=true
+         --spec_path=$(PWD)/exporters/$*/e2e/e2e_spec.yml --verbose_mode --agent_enabled=true
```
Comment (paologallinaharbur) [inline Makefile:54]: agent enabled if I am not mistaken is true by default 
https://github.com/newrelic/newrelic-integration-e2e-action/blob/5613eb6b876cec56756f7f2f08472d8904c5752d/newrelic-integration-e2e/cmd/main.go#L30

File: nri-config-generator/internal/config/config_test.go [hunk 3]
```diff
@@ -30,7 +30,7 @@ transformations:
 func TestGeneratorConfigPortAlreadyInUse(t *testing.T) {
 	dir := t.TempDir()
 	p := path.Join(dir, "config.yml")
-	err := ioutil.WriteFile(p, []byte(text), fs.ModePerm)
+	err := os.WriteFile(p, []byte(text), fs.ModePerm)
```
Comment (paologallinaharbur) [inline nri-config-generator/internal/config/config_test.go:33]: are this changes related to the PR?

Comment (kang-makes) [reply nri-config-generator/internal/config/config_test.go:33]: Semgrep complains of a deprecated API.

Comment (marcsanmi) [reply nri-config-generator/internal/config/config_test.go:33]: I addressed them in this separate commit b0ffe2617aa9fe317faa4fbe5c85f03bb6f40f77 as they were small changes.

File: Makefile (hunk 1 above)
Comment (marcsanmi) [reply Makefile:10]: I think until we fully fix the go run command yes... It's what I am doing right now. The `go run`  for any reason is pointing to an old version of the action, not the latest one.

File: Makefile (hunk 2 above)
Comment (marcsanmi) [reply Makefile]: Updated here ad5be98927005d66ed4be4a68d9abcd2786124a2.
